File: Client/java/apisdk/getdata/GetDataRequestBuilder.java
package apisdk.getdata;

import apisdk.getdata.sequence.SequenceRequestBuilder;
import apisdk.getdata.track.TrackRequestBuilder;
import com.microsoft.kiota.BaseRequestBuilder;
import com.microsoft.kiota.RequestAdapter;
import java.util.HashMap;
import java.util.Objects;
/**
 * Builds and executes requests for operations under /getData
 */
@jakarta.annotation.Generated("com.microsoft.kiota")
public class GetDataRequestBuilder extends BaseRequestBuilder {
    /**
     * The sequence property
     * @return a {@link SequenceRequestBuilder}
     */
    @jakarta.annotation.Nonnull
    public SequenceRequestBuilder sequence() {
        return new SequenceRequestBuilder(pathParameters, requestAdapter);
    }
    /**
     * The track property
     * @return a {@link TrackRequestBuilder}
     */
    @jakarta.annotation.Nonnull
    public TrackRequestBuilder track() {
        return new TrackRequestBuilder(pathParameters, requestAdapter);
    }
    /**
     * Instantiates a new {@link GetDataRequestBuilder} and sets the default values.
     * @param pathParameters Path parameters for the request
     * @param requestAdapter The request adapter to use to execute the requests.
     */
    public GetDataRequestBuilder(@jakarta.annotation.Nonnull final HashMap<String, Object> pathParameters, @jakarta.annotation.Nonnull final RequestAdapter requestAdapter) {
        super(requestAdapter, "{+baseurl}/getData", pathParameters);
    }
    /**
     * Instantiates a new {@link GetDataRequestBuilder} and sets the default values.
     * @param rawUrl The raw URL to use for the request builder.
     * @param requestAdapter The request adapter to use to execute the requests.
     */
    public GetDataRequestBuilder(@jakarta.annotation.Nonnull final String rawUrl, @jakarta.annotation.Nonnull final RequestAdapter requestAdapter) {
        super(requestAdapter, "{+baseurl}/getData", rawUrl);
    }
}

File: Client/ruby/get_data/get_data_request_builder.rb
require 'microsoft_kiota_abstractions'
require_relative '../api_sdk'
require_relative './get_data'
require_relative './sequence/sequence_request_builder'
require_relative './track/track_request_builder'

module ApiSdk
    module GetData
        ## 
        # Builds and executes requests for operations under \getData
        class GetDataRequestBuilder < MicrosoftKiotaAbstractions::BaseRequestBuilder
            
            ## 
            # The sequence property
            def sequence()
                return ApiSdk::GetData::Sequence::SequenceRequestBuilder.new(@path_parameters, @request_adapter)
            end
            ## 
            # The track property
            def track()
                return ApiSdk::GetData::Track::TrackRequestBuilder.new(@path_parameters, @request_adapter)
            end
            ## 
            ## Instantiates a new GetDataRequestBuilder and sets the default values.
            ## @param path_parameters Path parameters for the request
            ## @param request_adapter The request adapter to use to execute the requests.
            ## @return a void
            ## 
            def initialize(path_parameters, request_adapter)
                super(path_parameters, request_adapter, "{+baseurl}/getData")
            end
        end
    end
end
